### ISSUE TITLE
since v3.2.16: install of github located tag/version no longer possible (worked with v3.1.6)

### ISSUE THREAD

**Issue by HGlab01** [ioBroker/ioBroker.js-controller#1256]
With js-controller versions 3.1.6 (and older) an installation via the "cat" of a version/tag like
https://github.com/HGlab01/ioBroker.fuelpricemonitor/archive/v0.2.1-0.tar.gz
worked very fine! This was very helpful to share pre-releases for test-purpose to others.

Since js-controller version 3.2.16 this does no longer work :-(
```
2021-03-07 16:56:08.236 - info: host.ioBroker iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor/archive/v0.2.1-0.tar.gz"
2021-03-07 16:56:09.566 - info: host.ioBroker iobroker install HGlab01/ioBroker.fuelpricemonitor#v0.2.1-0.tar
2021-03-07 16:56:09.922 - info: host.ioBroker iobroker NPM version: 6.14.10
2021-03-07 16:56:09.923 - info: host.ioBroker iobroker npm install HGlab01/ioBroker.fuelpricemonitor#v0.2.1-0.tar --loglevel error --prefix "/opt/iobroker" (System call)
2021-03-07 16:56:14.998 - info: host.ioBroker iobroker npm ERR! code 1
2021-03-07 16:56:15.002 - info: host.ioBroker iobroker npm ERR! Command failed: git checkout v0.2.1-0.tar
2021-03-07 16:56:15.002 - info: host.ioBroker iobroker npm ERR! error: pathspec 'v0.2.1-0.tar' did not match any file(s) known to git.npm ERR!
2021-03-07 16:56:15.017 - info: host.ioBroker iobroker
2021-03-07 16:56:15.017 - info: host.ioBroker iobroker npm ERR! A complete log of this run can be found in:npm ERR! /home/iobroker/.npm/_logs/2021-03-07T15_56_15_003Z-debug.log
```

With js-controller v3.1.6 it worked fine:
```
2021-03-02 08:15:32.466 - info: host.ioBroker iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor/archive/v0.2.1-0.tar.gz"
2021-03-02 08:15:33.733 - info: host.ioBroker iobroker install https://github.com/HGlab01/ioBroker.fuelpricemonitor/archive/v0.2.1-0.tar.gz
2021-03-02 08:15:34.468 - info: host.ioBroker iobroker NPM version: 6.14.10npm install https://github.com/HGlab01/ioBroker.fuelpricemonitor/archive/v0.2.1-0.tar.gz --loglevel error --prefix "/opt/iobroker" (System call)
2021-03-02 08:15:52.551 - info: host.ioBroker iobroker upload [5] fuelpricemonitor.admin /opt/iobroker/node_modules/iobroker.fuelpricemonitor/admin/words.js words.js application/javascript
2021-03-02 08:15:52.629 - info: host.ioBroker iobroker upload [4] fuelpricemonitor.admin /opt/iobroker/node_modules/iobroker.fuelpricemonitor/admin/tsconfig.json tsconfig.json application/json
2021-03-02 08:15:52.685 - info: host.ioBroker iobroker upload [3] fuelpricemonitor.admin /opt/iobroker/node_modules/iobroker.fuelpricemonitor/admin/style.css style.css text/css
2021-03-02 08:15:52.742 - info: host.ioBroker iobroker upload [2] fuelpricemonitor.admin /opt/iobroker/node_modules/iobroker.fuelpricemonitor/admin/index_m.html index_m.html text/html
2021-03-02 08:15:52.800 - info: host.ioBroker iobroker upload [1] fuelpricemonitor.admin /opt/iobroker/node_modules/iobroker.fuelpricemonitor/admin/fuelpricemonitor.png fuelpricemonitor.png image/png
2021-03-02 08:15:52.873 - info: host.ioBroker iobroker upload [0] fuelpricemonitor.admin /opt/iobroker/node_modules/iobroker.fuelpricemonitor/admin/admin.d.ts admin.d.ts video/mp2t
2021-03-02 08:15:52.964 - info: host.ioBroker iobroker Update "system.adapter.fuelpricemonitor.0"
2021-03-02 08:15:52.982 - info: host.ioBroker stopInstance system.adapter.fuelpricemonitor.0 (force=false, process=false)
2021-03-02 08:15:52.983 - info: host.ioBroker stopInstance canceled schedule system.adapter.fuelpricemonitor.0
2021-03-02 08:15:53.996 - info: host.ioBroker iobroker exit 0
```
Is there an other solution to share pre-release in advance from a feature-branch without merging to master and/or npm publish?

Can be reproduced on linux, windows and Raspberry pi os.
NodeJS: v10.24.0 
Admin: v4.2.1

**Comment by Apollon77** [comment]
simply use the new variant to install a tag ... so just #v0.2.1-0 at the end ... the full tar.gz patch is not needed, but yes we can fix that

**Comment by HGlab01** [comment]
I was not aware of this cool feature like
https://github.com/HGlab01/ioBroker.fuelpricemonitor#v0.2.1-0
Thanks!

**Comment by Apollon77** [comment]
Reading changelog rulez ;-))

**Comment by HGlab01** [comment]
I tagged a pre-release on the feature branch
https://github.com/HGlab01/ioBroker.fuelpricemonitor/releases/tag/v0.2.2-0 (commit 4823f0b4d8916a23b15f28876b605a2748c334b8)
and tried to install the pre-release by using
iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor#0.2.2-0"

But installed was the latest commit on the master/main branch (c0fb11e4c91f5ed72a34086d028abcd1bc9bbdf3):

```
2021-03-12 13:55:52.398 - info: host.LAPTOP-N0KJFISK iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor#0.2.2-0"
2021-03-12 13:55:54.383 - info: host.LAPTOP-N0KJFISK iobroker install HGlab01/ioBroker.fuelpricemonitor#c0fb11e4c91f5ed72a34086d028abcd1bc9bbdf3
2021-03-12 13:55:55.124 - info: host.LAPTOP-N0KJFISK iobroker NPM version: 6.14.4
2021-03-12 13:55:55.125 - info: host.LAPTOP-N0KJFISK iobroker npm install HGlab01/ioBroker.fuelpricemonitor#c0fb11e4c91f5ed72a34086d028abcd1bc9bbdf3 --loglevel error (System call)
```

Same behavior if I take the commit ID corresponding to the tag
```
2021-03-12 13:59:54.244 - info: host.LAPTOP-N0KJFISK iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor#4823f0b4d8916a23b15f28876b605a2748c334b8"
2021-03-12 13:59:56.698 - info: host.LAPTOP-N0KJFISK iobroker install HGlab01/ioBroker.fuelpricemonitor#c0fb11e4c91f5ed72a34086d028abcd1bc9bbdf3
2021-03-12 13:59:58.268 - info: host.LAPTOP-N0KJFISK iobroker NPM version: 6.14.4
2021-03-12 13:59:58.269 - info: host.LAPTOP-N0KJFISK iobroker npm install HGlab01/ioBroker.fuelpricemonitor#c0fb11e4c91f5ed72a34086d028abcd1bc9bbdf3 --loglevel error (System call)
```

I also tried to use an older tag (which is available on the master/main as well). Same result: c0fb11e4c91f5ed72a34086d028abcd1bc9bbdf3 was installed

What I am doing wrong or is there a bug?
I am still using
Js-controller: v3.2.16
NodeJS: v10.24.0
Admin: v4.2.1

**Comment by HGlab01** [comment]
Hi @Apollon77,
any idea why it is not working by using #versionnumber as described above? Thanks!

**Comment by Apollon77** [comment]
hehe I now checked it again and in fact ... your tag is NOT "0.2.2-0" BUT "v0.2.2-0" ... does it work then?!

**Comment by HGlab01** [comment]
Hi  @Apollon77 again!
No it does not
Commit **b2bd0ee929fb951a948b66bee9f67e4c59f9b459** is used but v0.2.2-0 belongs to **4823f0b4d8916a23b15f28876b605a2748c334b8** (https://github.com/HGlab01/ioBroker.fuelpricemonitor/releases/tag/v0.2.2-0)

**b2bd0ee929fb951a948b66bee9f67e4c59f9b459** is the last commit on master.

```
2021-03-21 19:35:17.412 - info: host.LAPTOP-N0KJFISK iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0"
2021-03-21 19:35:19.736 - info: host.LAPTOP-N0KJFISK iobroker install HGlab01/ioBroker.fuelpricemonitor#b2bd0ee929fb951a948b66bee9f67e4c59f9b459
2021-03-21 19:35:21.056 - info: host.LAPTOP-N0KJFISK iobroker NPM version: 6.14.4
2021-03-21 19:35:21.057 - info: host.LAPTOP-N0KJFISK iobroker npm install HGlab01/ioBroker.fuelpricemonitor#b2bd0ee929fb951a948b66bee9f67e4c59f9b459 --loglevel error (System call)
```
Even if I use the commit **4823f0b4d8916a23b15f28876b605a2748c334b8** directly instead of the tag it does not work

```
2021-03-21 19:43:15.309 - info: host.LAPTOP-N0KJFISK iobroker url "https://github.com/HGlab01/ioBroker.fuelpricemonitor#4823f0b4d8916a23b15f28876b605a2748c334b8"
2021-03-21 19:43:17.728 - info: host.LAPTOP-N0KJFISK iobroker install HGlab01/ioBroker.fuelpricemonitor#b2bd0ee929fb951a948b66bee9f67e4c59f9b459
2021-03-21 19:43:18.977 - info: host.LAPTOP-N0KJFISK iobroker NPM version: 6.14.4
2021-03-21 19:43:18.979 - info: host.LAPTOP-N0KJFISK iobroker npm install HGlab01/ioBroker.fuelpricemonitor#b2bd0ee929fb951a948b66bee9f67e4c59f9b459 --loglevel error (System call)
```

**Comment by Apollon77** [comment]
@AlCalzone maybe you can have a look when you find time

**Comment by AlCalzone** [comment]
The problem here was that we didn't correctly detect .tar.gz as a file extension, only .zip and .gz. So `https://github.com/HGlab01/ioBroker.fuelpricemonitor/archive/v0.2.1-0.tar.gz` got resolved to the tag `v0.2.1-0.tar` instead of `v0.2.1-0`.

The other URLs are not valid:
-> `https://github.com/HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0` should be `HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0` instead without the URL prefix. This is a specifier for npm which automatically gets resolved to the correct commit on github.

**Comment by HGlab01** [comment]
Hi @AlCalzone,
thank you for your support!

`HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0` works:
```
2021-03-24 22:22:10.325 - info: host.ioBroker iobroker url "HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0"
2021-03-24 22:22:11.646 - info: host.ioBroker iobroker install HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0
2021-03-24 22:22:12.034 - info: host.ioBroker iobroker NPM version: 6.14.10
2021-03-24 22:22:12.035 - info: host.ioBroker iobroker npm install HGlab01/ioBroker.fuelpricemonitor#v0.2.2-0 --loglevel error --prefix "/opt/iobroker" (System call)

```
